### ISSUE TITLE
[Feature request] Global key event filtering

### ISSUE THREAD

**Issue by youdz** [angular/angular#7308]
When having a component that wants to listen to global key events, using `(body:keyup.escape)` crashes with an `EXCEPTION: not implemented`, whereas `(body:keyup)` works fine. Is this actually being worked on, or just not in the roadmap at all?

Here is my use case: I'm trying to build a modal component, and I want to dismiss it when the user presses escape. And any listener tied to the modal itself never triggers, so I need to listen to the keyup on `body`.

``` typescript
@Component({
    selector: "my-modal",
    host: {
        "(body:keyup.escape)": "dismiss()"
    }
    ...
})
export class MyModal {
    @Input("MyModalOpen") open = false;
    @Output("MyModalOpenChange") closed = new EventEmitter<boolean>();
    ...
}
```

The only solution I see right now is catch every keyup and filter myself, which feels dirty. I'm open to any alternative solutions. :smile: 
Thanks for your help.

**Comment by zoechi** [comment]
`window:xxx` worked for me. I haven't tried `body` yet.

**Comment by youdz** [comment]
`window:xxx` and `body:xxx` both work. The issue comes from `window:keyup.xxx` and `body:keyup.xxx`, the key filtering is not implemented. Sorry if that wasn't clear in my initial post.

**Comment by ericmdantas** [comment]
Yeah, I got this the other day. It'd be great to have it implemented.

**Comment by PascalPrecht** [comment]
> The only solution I see right now is catch every keyup and filter myself

@youdz that's correct. You'd rather listen to `keydown` event and then check the `keyCode` manually like this:

```
@Directive({
  selector: '[modalContainer]'
  host: {
    '(body:keydown)': 'documentKeypress($event)'
  }
})
class ModalContainer {
  ...
  documentKeypress(event: KeyboardEvent) {
    if (event.keyCode == KeyCodes.ESCAPE) {
      this.dialogRef.close();
    }
  }
}
```

**Comment by youdz** [comment]
Thanks Pascal, that's more or less what I had slapped together in the meantime.

I'll stick to it for now, but I renamed this issue into a feature request, because I don't think I'm the only one who would appreciate this. It seems like a low-hanging fruit in terms of feature added per amount of code, since you already have key filtering on non-global events. I haven't had time to look at the implementation of `platform/dom/events/` yet, but if I end up with a little free time and it hasn't been solved, I'll try to submit a PR myself.
